src/Microsoft.OData.Core/UriParser/Binders/InnerPathTokenBinder.cs

```diff
@@ -143,6 +144,15 @@ internal QueryNode BindInnerPathSegment(InnerPathToken segmentToken)
                 throw new ODataException(ODataErrorStrings.MetadataBinder_PropertyAccessSourceNotSingleValue(segmentToken.Identifier));
             }
 
+            if (collectionParent != null)
+            {
+                var parentType = collectionParent.EntityItemType;
+                var collectionProperty = this.Resolver.ResolveProperty(parentType.StructuredDefinition(), segmentToken.Identifier);
+
+                // TODO: do multiplicity and other checks like in GeneratePropertyAccessQueryNode method.
+                return new AggregatedCollectionPropertyNode(collectionParent, collectionProperty);

```

Comment:
>AggregatedCollectionPropertyNode [](start = 27, length = 32)

InnerPathTokenBinderShouldFailIfPropertySourceIsNotASingleValue fails because the AggregatedCollectionPropertyNode throws an exception; collectionProperty in this case in a navigation property and not a structural property.

Ideally, we could check for this and return an OData exception instead of the ArgumentException being generated by AggregatedCollectionPropertyNode.